utils.js

```diff
@@ -119,6 +147,10 @@ let utils = {
                 token: options.refreshToken,
                 xthorn: options.xthorn,
             }).then((response) => {
+                if (!utils.isSuccessfulResponse(response.response.statusCode)) {
```

Comment:
Note that `_oauthRequest` already calls `utils.assertSaneResponse`. There's no need to call it twice.